app/models/work.rb

```diff
@@ -183,10 +183,34 @@ def new_recipients_allow_gifts
     self.new_gifts.each do |gift|
       next if gift.pseud.blank?
       next if gift.pseud&.user&.preference&.allow_gifts?
-      next if self.challenge_assignments.map(&:requesting_pseud).include?(gift.pseud)
-      next if self.challenge_claims.reject { |c| c.request_prompt.anonymous? }.map(&:requesting_pseud).include?(gift.pseud)
+      next if challenge_bypass(gift)
 
-      self.errors.add(:base, ts("%{byline} does not accept gifts.", byline: gift.pseud.byline))
+      self.errors.add(:base, I18n.t("activerecord.errors.models.work.blocked_gifts", byline: gift.pseud.byline))
```

Comment:
This (and the other error lines below) should not have the explicit I18n.t call. The [errors.add](https://rubydoc.info/gems/activemodel/6.1.7.4/ActiveModel/Errors#add-instance_method) method will translate the symbol using the `activerecord.errors.models` scope, see [generate_message](https://rubydoc.info/gems/activemodel/6.1.7.4/ActiveModel/Errors#generate_message-instance_method).
So the code I suggested earlier (`errors.add(:base, :blocked_gifts, byline: gift.pseud.byline)`) should work as is.

#4488 is an example of a translated error message like this.

Reply:
No issue. I fear the PHP/JS dev in me will never truly get used to Ruby magic. Even implicit returns at the end of functions still puzzle me sometimes. :sweat_smile: